shotover/src/server.rs

```diff
@@ -255,6 +263,133 @@ pub struct Handler<C: CodecBuilder> {
+        loop {
+            tokio::select! {
+                result = reader.next() => {
+                    if let Some(ws_message) = result {
+                        match ws_message {
+                            Ok(WsMessage::Binary(ws_message_data)) => {
+                                // Entire message is reallocated and copied here due to incompatibility between tokio codecs and tungstenite.
+                                let message = decoder.decode(&mut BytesMut::from(ws_message_data.as_slice()));
+                                match message {
+                                    Ok(Some(message)) => {
+                                        if in_tx.send(message).is_err() {
+                                            // main task has shutdown, this task is no longer needed
+                                            return;
+                                        }
+                                    }
+                                    Ok(None) => {
+                                        // websocket client has closed the connection
+                                        return;
+                                    }
+                                    Err(CodecReadError::RespondAndThenCloseConnection(messages)) => {
+                                        if let Err(err) = out_tx.send(messages) { // TODO we need
+                                                                                  // to send a
+                                                                                  // close message
+                                                                                  // to the client
+                                            error!("Failed to send RespondAndThenCloseConnection message: {:?}", err);
+                                        }
+                                        return;
+                                    }
+                                    Err(CodecReadError::Parser(err)) => { // TODO we need to send a
+                                                                          // close message to the
+                                                                          // client, protocol error
+                                        warn!("failed to decode message: {:?}", err);
+                                        return;
+                                    }
+                                    Err(CodecReadError::Io(_err)) => {
+                                        unreachable!("CodecReadError::Io should not occur because we are reading from a newly created BytesMut")
+                                    }
+                                }
+                            }
+                            Ok(_ws_message) => {
+                                // TODO we need to tell the client about a protocol error
+
+                                todo!();
+                            }
+                            Err(err) => panic!("{err}") // TODO
```

Comment:
Could we get rid of these todo / panics on the match? even if it just bails on the connection altogether and logs it instead of killing the whole shotover process